src/include/grpc_transcoding/path_matcher.h

```diff
@@ -113,6 +115,22 @@ class PathMatcherBuilder {
   bool Register(const std::string& http_method, const std::string& path,
                 const std::string& body_field_path, Method method);
 
+  // When set to true, URL path parameters will be fully URI-decoded.
+  //
+  // The default behavior is to not decode RFC 6570 reserved characters in multi
+  // segment matches.
+  void SetAlwaysDecode(bool value) { always_decode_ = value; }
+
+  // When set to true, URL path parameters will be fully URI-decoded except in
+  // cases of single segment matches in reserved expansion, where "%2F" will be
+  // left encoded.
+  //
+  // The default behavior is to not decode RFC 6570 reserved characters in multi
+  // segment matches.
+  void SetFullyDecodeReservedExpansion(bool value) {
```

Comment:
1) I like the names to be more specific for readability.  how about  `UrlUnescapeSpec`?

2) These two bools are mutual exclusive,  Can we combine them into one flag with an enum value?


       
}

Reply:
Thanks, updated